geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Stat.scala

```diff
@@ -0,0 +1,173 @@
+}
+
+/**
+ * This class contains parsers which dictate how to instantiate a particular Stat.
+ * Stats are created by passing a stats string as a query hint (QueryHints.STATS_STRING).
+ *
+ * A valid stats string should adhere to the parsers here:
+ * e.g. "MinMax(attributeName);IteratorCount" or "RangeHistogram(attributeName,10,0,100)"
+ * (see tests for more use cases)
+ */
+object Stat {
+  class StatParser(sft: SimpleFeatureType) extends RegexParsers {
+    val attributeNameRegex = """\w+""".r
+    val numBinRegex = """[1-9][0-9]*""".r // any non-zero positive int
+    val nonEmptyRegex = """[^,)]+""".r
+    val dateFormat = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+
+    /**
+     * Obtains the index of the attribute within the SFT
+     *
+     * @param attribute attribute name as a string
+     * @return attribute index
+     */
+    private def getAttrIndex(attribute: String): Int = {
+      val attrIndex = sft.indexOf(attribute)
+      if (attrIndex == -1)
+        throw new Exception(s"Invalid attribute name in stat string: $attribute")
+      attrIndex
+    }
+
+    def minMaxParser: Parser[MinMax[_]] = {
+      "MinMax(" ~> attributeNameRegex <~ ")" ^^ {
+        case attribute =>
+          val attrIndex = getAttrIndex(attribute)
+          val attrType = sft.getType(attribute).getBinding
+          val attrTypeString = attrType.getName
+          attrType match {
+            case _ if attrType == classOf[Date] =>
+              new MinMax[Date](attrIndex, attrTypeString, new Date(java.lang.Long.MAX_VALUE), new Date(java.lang.Long.MIN_VALUE))
+            case _ if attrType == classOf[java.lang.Integer] =>
+              new MinMax[java.lang.Integer](attrIndex, attrTypeString, Integer.MAX_VALUE, Integer.MIN_VALUE)
+            case _ if attrType == classOf[java.lang.Long] =>
+              new MinMax[java.lang.Long](attrIndex, attrTypeString, java.lang.Long.MAX_VALUE, java.lang.Long.MIN_VALUE)
+            case _ if attrType == classOf[java.lang.Float] =>
+              new MinMax[java.lang.Float](attrIndex, attrTypeString, java.lang.Float.MAX_VALUE, java.lang.Float.MIN_VALUE)
+            case _ if attrType == classOf[java.lang.Double] =>
+              new MinMax[java.lang.Double](attrIndex, attrTypeString, java.lang.Double.MAX_VALUE, java.lang.Double.MIN_VALUE)
+            case _ =>
+              throw new Exception(s"Cannot create stat for invalid type: $attrType for attribute: $attribute")
+          }
+      }
+    }
+
+    def iteratorStackParser: Parser[IteratorStackCounter] = {
+      "IteratorStackCounter" ^^ { case _ => IteratorStackCounter() }
+    }
+
+    def enumeratedHistogramParser: Parser[EnumeratedHistogram[_]] = {
+      "EnumeratedHistogram(" ~> attributeNameRegex <~ ")" ^^ {
+        case attribute =>
+          val attrIndex = getAttrIndex(attribute)
+          val attrType = sft.getType(attribute).getBinding
+          val attrTypeString = attrType.getName
+          attrType match {
```

Comment:
I just meant instead of ```MinMax("foo")``` use ```MinMax("foo:String")```. You'd still have to convert from a string to a class, but you wouldn't need the sft for it.